client/VDMSClientImpl.h

```diff
@@ -0,0 +1,87 @@
+/**
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2017 Intel Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify,
+ * merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+#pragma once
+
+#include <chrono>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "Protocol.h"
+#include "VDMSClient.h"
+
+namespace VDMS {
+
+    struct AuthToken {
+        std::chrono::system_clock::time_point issued_at;
+        std::string refresh_token;
+        int32_t refresh_token_expires_in{-1};
```

Comment:
I'll leave the name change to Luis.

As for the offset type, I did use a `time_point` at first but decided to move to a signed integer because I didn't come up with a **clean** way to represent a null/undefined/invalid value. If we were compiling with c++17 I could have used `std::optional`.

Reply:
we don't have a convention/rule for this. We only include the unit when it is not more or less obvious what the unit is, mainly to avoid having longer and redundant variable names. When the units are different from expected within a context, we tend to add it, or at least add a comment on the header file. In this case (and in the case of BYTES marked below), the units are more or less what it is expected within the context.